src/main/java/uk/gov/hmcts/ccd/domain/model/common/CaseFieldPathUtils.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.hmcts.ccd.domain.model.common;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang3.StringUtils;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CommonField;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseFieldDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.FieldTypeDefinition;
+
+import java.util.*;
+
+import static java.util.stream.Collectors.toList;
+
+public class CaseFieldPathUtils {
+
+    private static final String SEPARATOR = ".";
+    private static final String SEPARATOR_REGEX = "\\.";
+
+    private CaseFieldPathUtils() {
+    }
+
+    public static Optional<CommonField> getFieldDefinitionByPath(CaseTypeDefinition caseTypeDefinition, String path) {
+        if (StringUtils.isBlank(path)) {
+            return Optional.empty();
+        }
+        List<String> pathElements = getPathElements(path);
+
+        Optional<CaseFieldDefinition> topLevelCaseField = caseTypeDefinition.getCaseField(pathElements.get(0));
+
+        return topLevelCaseField.flatMap(field -> getFieldDefinitionByPath(field, getPathElementsTailAsString(pathElements)));
+    }
+
+    public static Optional<CommonField> getFieldDefinitionByPath(CommonField commonField, String path) {
+        if (StringUtils.isBlank(path)) {
+            return Optional.of(commonField);
+        }
+
+        return getFieldDefinitionByPath(commonField.getFieldTypeDefinition(), path, false);
+    }
+
+    public static Optional<CommonField> getFieldDefinitionByPath(FieldTypeDefinition fieldTypeDefinition,
+                                                                 String path,
+                                                                 boolean pathIncludesParent) {
+        if (StringUtils.isBlank(path) || fieldTypeDefinition.getChildren().isEmpty() || (pathIncludesParent && splitPath(path).length == 1)) {
+            return Optional.empty();
+        }
+        List<String> pathElements = getPathElements(path);
+
+        return reduce(fieldTypeDefinition.getChildren(), pathIncludesParent ? getPathElementsTail(pathElements) : pathElements);
+    }
+
+    public static JsonNode getCaseFieldNodeByPath(JsonNode node, String path) {
```

Comment:
can we find a better name? maybe getNestedCaseFieldByPath?
can you also add a little javadoc with description here and on the other methods too please